### ISSUE TITLE
Bugs in Notebook

### ISSUE THREAD

**Issue by vsoch** [LLNL/weave-demos#1]
Hiya! I was able to get everything running with maestro in a container, and I added a final step to dynamically render the notebook, and ran into a few bugs I'd like to report / discuss here.

The first was an issue with a pydv import - I opened an issue and PR to fix it, and my workaround for the time being is to use a fixed version of the module. https://github.com/LLNL/PyDV/pull/276

The second was that the group name was hard coded - I decided instead to just choose the first from the set:

```python
# Second issue - hard coded group
# For the purposes of this demo, chose the first.
GROUP_OF_INTEREST = list(groups)[0]
print(f"Group of interest is {GROUP_OF_INTEREST}")
```

The third was that the plotting seems to be outdated (or I suspect the TODO suggests it wasn't implemented yet?) "curves" is a list, and it doesn't have a function `create_curve` so I commented it out:

```python
# Commented out - this returns a list and does not work
# Likely it's raw data we just need to plot with matplotlib
curves = pydv.readsina(PYDV_DEMO_FILENAME)  # raw?
print(curves)
# curve = curves.create_curve(x="time", y="y_pos")
# plt.plot(pydv.fft(curve))
```

Finally (this one I didn't dive into debugging) there is a key error:

```python
KeyError                                  Traceback (most recent call last)
Cell In[1], line 23
     21 datastore = create_datastore(database)
     22 recs = datastore.records
---> 23 vis = Visualizer(datastore)
     24 print("Sina is ready!")

File /usr/local/lib/python3.8/dist-packages/sina/visualization.py:86, in Visualizer.__init__(self, ds, id_pool, matplotlib_options)
     84 self.data_names = self.get_contained_data_names()
     85 # Used for histograms and the like
---> 86 self.data_names["scalar_and_string"] = sorted(set(self.data_names["scalar"])
     87                                               .union(self.data_names["string"]))
     88 self.id_pool = id_pool
     89 self.matplotlib_options = matplotlib_options

KeyError: 'scalar'
KeyError: 'scalar'
```

Help with the above would be great! The container I have as a WIP (pending these fixes) is here: https://github.com/rse-ops/flux-hpc/pull/23. Once these updates are in it should be fairly trivial to merge, that will trigger an automated build to GitHub packages, and then I can throw together a test recipe to see how Maestro performs on the flux operator. My one concern is that Maestro doesn't have a single run command that will continue (and not exit) while everything is running, so we will need to use an interactive submit, and then wait for the output files to appear as an indication that everything is done. If you know maestro, I also couldn't figure out how to use `matestro status` (sorry was my first time!)

Thanks for the help!

**Comment by doutriaux1** [comment]
Thanks @vsoch I believe some of these are fixed in some other branches.

**Comment by doutriaux1** [comment]
@vsoch also please take a look at the trilab-demo branch it's a bit more up to date.

**Comment by HaluskaR** [comment]
Great catch on the last one! That KeyError: 'scalar' is caused by the visualizer expecting there to be scalar data, but finding none. Most of the time this is unintentional (setting up a visualizer against an empty database) since most dbs have some scalar data in them. That said, absolutely shouldn't throw an error, should print a warning at most. I'll get on it!

**Comment by vsoch** [comment]
Thanks @doutriaux1 and @HaluskaR - are there plans to merge the trilabs branch into main so folks can easily use the latest?

**Comment by doutriaux1** [comment]
@vsoch yes we will be merging a few branches soon (on gitlab) and bring it back here.

**Comment by vsoch** [comment]
heyo! Wanted to circle back and see if there are any updates? I can add the current (partial workflow) as an example to the Flux Operator, but I'd love to have the whole thing working fully as expected. Thank you!

**Comment by HaluskaR** [comment]
Fix for the last one's in! https://lc.llnl.gov/gitlab/siboka/Sina/-/merge_requests/219 

This'll replace the ValueError a warning, but I did discover that the path causing the error is for a truly empty database. Given that the snippet appears to be connecting to an existing database, it might be worth double-checking that path.

**Comment by vsoch** [comment]
Does the existing database get generated by the first notebook of the tutorial (and thus this is the path I should check) is it expected to be provided by the knowledgeable user?

If the first - I can double check my path!
